common/src/main/java/com/microsoft/identity/common/internal/fido/WebAuthnJsonUtil.kt

```diff
@@ -67,17 +71,25 @@ class WebAuthnJsonUtil {
         /**
          * Extracts the AuthenticatorAssertionResponse from the overall AuthenticationResponse string received from the authenticator.
          * @param fullResponseJson AuthenticationResponse Json string.
+         * @throws JSONException if a value is not present that should be.
          */
         fun extractAuthenticatorAssertionResponseJson(fullResponseJson : String): String {
             val fullResponseJsonObject = JSONObject(fullResponseJson);
-            var authResponseJsonObject = fullResponseJsonObject
+            val authResponseJsonObject = fullResponseJsonObject
                 .getJSONObject(FidoConstants.WEBAUTHN_AUTHENTICATION_ASSERTION_RESPONSE_JSON_KEY)
-            // Making sure that Id is here because ESTS expects it.
-            // I've noticed that GPM will sometimes not include the id in the response object.
-            if (!authResponseJsonObject.has(WEBAUTHN_RESPONSE_ID_JSON_KEY)) {
-                authResponseJsonObject = authResponseJsonObject.put(WEBAUTHN_RESPONSE_ID_JSON_KEY, fullResponseJsonObject.get(WEBAUTHN_RESPONSE_ID_JSON_KEY))
-            }
-            return authResponseJsonObject.toString()
+            // ESTS expects a custom object with clientDataJSON, authenticatorData, signature, userHandle, and id.
+            val assertionResult = JSONObject();
+            assertionResult.put(WEBAUTHN_RESPONSE_ID_JSON_KEY, fullResponseJsonObject.get(
```

Comment:
I'm assuming you're already handling the case where these get() returns null.

(if some of the fields return null, do we still send this response to ESTS? or do we send an error response, or do we fail here?, etc)

Reply:
It will throw a JSONException, which will be caught by a catch clause in AuthFidoChallengeHandler, and that exception will be sent back to ESTS via the protocol.